### PR TITLE
Feature - notification polling

### PR DESCRIPTION
**Pull request by joefitter**
* Added notifications table for polling notifications

### CODE REVIEW

File: migrations/20210401161450_add_notifications_table.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  return knex.schema
+    .createTable('notifications', table => {
+      table.uuid('id').primary().defaultTo(knex.raw('uuid_generate_v4()'));
+      table.string('to').notNull();
+      table.string('name').notNull();
+      table.string('subject').notNull();
+      table.text('html').notNull();
+      table.string('identifier').notNull();
+      table.dateTime('completed').nullable();
+      table.dateTime('deleted');
```
Comment (wheelsandcogs) [inline migrations/20210401161450_add_notifications_table.js:12]: Should this also be `nullable()` (maybe that's the default?)

File: schema/notification.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return 'notifications';
+  }
+
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      additionalProperties: false,
+      properties: {
+        id: { type: 'string', pattern: uuid.v4 },
+        to: { type: 'string' },
```
Comment (wheelsandcogs) [inline schema/notification.js:15]: Do we want to enforce email format here, as in https://github.com/UKHomeOffice/asl-schema/blob/master/schema/profile.js#L76

File: migrations/20210401161450_add_notifications_table.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+exports.up = function(knex) {
+  return knex.schema
+    .createTable('notifications', table => {
+      table.uuid('id').primary().defaultTo(knex.raw('uuid_generate_v4()'));
+      table.string('to').notNull();
```
Comment (wheelsandcogs) [inline migrations/20210401161450_add_notifications_table.js:6]: might want an index on this.

File: migrations/20210401161450_add_notifications_table.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      table.string('name').notNull();
+      table.string('subject').notNull();
+      table.text('html').notNull();
+      table.string('identifier').notNull();
```
Comment (wheelsandcogs) [inline migrations/20210401161450_add_notifications_table.js:10]: and probably an index on this as well.

File: migrations/20210401161450_add_notifications_table.js (hunk 3 above)
Comment (wheelsandcogs) [reply migrations/20210401161450_add_notifications_table.js:6]: Do we always have profile_id? I'd be tempted to stick a column for that in as well.

File: schema/notification.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        html: { type: 'string' },
+        identifier: { type: 'string' },
+        completed: { type: ['string', 'null'], format: 'date-time' },
+        createdAt: { type: 'string', format: 'date-time' },
+        updatedAt: { type: 'string', format: 'date-time' },
+        deleted: { type: ['string', 'null'], format: 'date-time' }
+      }
+    };
+  }
+
```
Comment (wheelsandcogs) [inline schema/notification.js:27]: Would be nice to have a `relationMapping` for profiles and vice versa.

File: migrations/20210401161450_add_notifications_table.js (hunk 1 above)
Comment (lennym) [reply migrations/20210401161450_add_notifications_table.js:12]: It is the default.

File: migrations/20210401161450_add_notifications_table.js (hunk 4 above)
Comment (lennym) [reply migrations/20210401161450_add_notifications_table.js:10]: Definitely an index on this.

**Comment by lennym** [comment]
I agree with Phil's suggestion of a profile id column, for the reason that if we want to add in-system display of the same notifications then it will be trivial to query this table, as well as adding resilience to changing email addresses.

File: schema/notification.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      additionalProperties: false,
+      properties: {
+        id: { type: 'string', pattern: uuid.v4 },
+        profileId: { type: 'string', pattern: uuid.v4 },
+        to: { type: 'string' },
```
Comment (wheelsandcogs) [inline schema/notification.js:16]:
```suggestion
        to: { type: 'string', format: 'email' },
```

**Pull request merged**